### PR TITLE
Integrate external network with peering

### PR DESCRIPTION
**Pull request by aleoli**
# Description

This pr integrates the creation of the external network inside the peering process

# How Has This Been Tested?

- [x] locally

### CODE REVIEW

**Review by cheina97 — APPROVED** [review]

File: pkg/liqo-controller-manager/foreign-cluster-operator/externalnetwork.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		return false, err
+	}
+
+	switch len(gwClients.Items) {
+	case 0:
+	case 1:
+		return true, nil
+	default:
+		return true, fmt.Errorf("multiple GatewayClients found for ForeignCluster %s", fc.Spec.ClusterIdentity.ClusterID)
+	}
```
Comment (cheina97) [inline pkg/liqo-controller-manager/foreign-cluster-operator/externalnetwork.go:135]: Can you move it in pkg/utils/getters/k8sGetters.go ?

File: pkg/liqo-controller-manager/foreign-cluster-operator/externalnetwork.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+			ownerRef := gwServers.Items[0].OwnerReferences[i]
+			if ownerRef.Kind == networkingv1alpha1.ExternalNetworkKind {
+				// the GatewayServer is managed by the ExternalNetwork, so we can enforce it by ignoring its existence
+				return false, nil
+			}
+		}
+		return true, nil
+	default:
+		return true, fmt.Errorf("multiple GatewayServers found for ForeignCluster %s", fc.Spec.ClusterIdentity.ClusterID)
+	}
```
Comment (cheina97) [inline pkg/liqo-controller-manager/foreign-cluster-operator/externalnetwork.go:120]: Can you move it in pkg/utils/getters/k8sGetters.go ?

File: cmd/liqo-controller-manager/main.go [hunk 3]
```diff
@@ -227,6 +228,12 @@ func main() {
 	flag.StringVar(&wgGatewayClientClusterRoleName, "wg-gateway-client-cluster-role-name", "liqo-gateway",
 		"The name of the cluster role used by the wireguard gateway clients")
 
+	// Gateway parameters
+	gatewayServiceType := flag.String("gateway-service-type", "LoadBalancer", "The type of the gateway service")
+	gatewayServicePort := flag.Int("gateway-service-port", 51820, "The port of the gateway service")
+	gatewayMTU := flag.Int("gateway-mtu", 1450, "The MTU of the gateway interface")
```
Comment (fra98) [inline cmd/liqo-controller-manager/main.go:234]:
```suggestion
	gatewayMTU := flag.Int("gateway-mtu", 1340, "The MTU of the gateway interface")
```

File: pkg/liqo-controller-manager/external-network/externalnetwork-controller/externalnetwork_controller.go [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+	if remoteClusterID == "" {
+		return nil, fmt.Errorf("no remote ClusterID found for ExternalNetwork %q", extNet.Name)
+	}
+
+	opts := &gatewayclient.ForgeOptions{
+		KubeClient:        r.KubeClient,
+		RemoteClusterID:   remoteClusterID,
+		GatewayType:       gatewayclient.DefaultGatewayType,
+		TemplateName:      gatewayclient.DefaultTemplateName,
+		TemplateNamespace: r.LiqoNamespace,
```
Comment (fra98) [inline pkg/liqo-controller-manager/external-network/externalnetwork-controller/externalnetwork_controller.go:279]:
```suggestion
		TemplateNamespace: gatewayclient.DefaultTemplateNamespace,
```

File: pkg/liqo-controller-manager/external-network/externalnetwork-controller/externalnetwork_controller.go [hunk 5]
```diff
@@ -0,0 +1,414 @@
...
+	if remoteClusterID == "" {
+		return nil, fmt.Errorf("no remote ClusterID found for ExternalNetwork %q", extNet.Name)
+	}
+
+	opts := &gatewayserver.ForgeOptions{
+		KubeClient:        r.KubeClient,
+		RemoteClusterID:   remoteClusterID,
+		GatewayType:       gatewayserver.DefaultGatewayType,
+		TemplateName:      gatewayserver.DefaultTemplateName,
+		TemplateNamespace: r.LiqoNamespace,
```
Comment (fra98) [inline pkg/liqo-controller-manager/external-network/externalnetwork-controller/externalnetwork_controller.go:338]:
```suggestion
		TemplateNamespace: gatewayserver.DefaultTemplateNamespace,
```

Comment (aleoli) [reply pkg/liqo-controller-manager/external-network/externalnetwork-controller/externalnetwork_controller.go:338]: This value is received by a flag indicating where liqo is installed; I think it's safer than the variable

**Pull request merged**